pyomo/contrib/doe/doe.py

```diff
@@ -25,1255 +25,1169 @@
-            for j, un in enumerate(m.regression_parameters):
-                # Copy value from Pyomo model into numpy array
-                fim[i][j] = m.fim[bu, un].value
+        # Make objective block for constraints connected to objective
+        model.obj_cons = pyo.Block()
 
-                # Set lower bound to ensure diagonal elements are (almost) non-negative
-                # if i == j:
-                #     m.fim[bu, un].setlb(-small_number)
+        # Assemble the FIM matrix. This is helpful for initialization!
+        fim_vals = [
+            model.fim[bu, un].value
+            for i, bu in enumerate(model.parameter_names)
+            for j, un in enumerate(model.parameter_names)
+        ]
+        fim = np.array(fim_vals).reshape(
+            len(model.parameter_names), len(model.parameter_names)
+        )
 
         ### Initialize the Cholesky decomposition matrix
-        if self.Cholesky_option and self.objective_option == ObjectiveLib.det:
+        if self.Cholesky_option and self.objective_option == ObjectiveLib.determinant:
 
             # Calculate the eigenvalues of the FIM matrix
             eig = np.linalg.eigvals(fim)
 
             # If the smallest eigenvalue is (practically) negative, add a diagonal matrix to make it positive definite
             small_number = 1e-10
             if min(eig) < small_number:
-                fim = fim + np.eye(len(self.param)) * (small_number - min(eig))
+                fim = fim + np.eye(len(model.parameter_names)) * (
+                    small_number - min(eig)
+                )
 
             # Compute the Cholesky decomposition of the FIM matrix
             L = np.linalg.cholesky(fim)
 
             # Initialize the Cholesky matrix
-            for i, c in enumerate(m.regression_parameters):
-                for j, d in enumerate(m.regression_parameters):
-                    m.L_ele[c, d].value = L[i, j]
+            for i, c in enumerate(model.parameter_names):
+                for j, d in enumerate(model.parameter_names):
+                    model.L[c, d].value = L[i, j]
 
         def cholesky_imp(m, c, d):
             """
             Calculate Cholesky L matrix using algebraic constraints
             """
-            # If it is the left bottom half of L
-            if list(self.param.keys()).index(c) >= list(self.param.keys()).index(d):
-                return m.fim[c, d] == sum(
-                    m.L_ele[c, list(self.param.keys())[k]]
-                    * m.L_ele[d, list(self.param.keys())[k]]
-                    for k in range(list(self.param.keys()).index(d) + 1)
+            # If the row is greater than or equal to the column, we are in the
+            # lower triangle region of the L and FIM matrices.
+            # This region is where our equations are well-defined.
+            if list(model.parameter_names).index(c) >= list(
+                model.parameter_names
+            ).index(d):
```

Comment:
Using a model object in this rule coming from outside of this function scope. You should use `m` here but remember that you are adding this constraint on a block, `model.obj_cons` so the block is what will be passed in as the first argument.